packages/grafana-ui/src/components/Toggletip/Toggletip.tsx

```diff
@@ -0,0 +1,301 @@
+}
+
+export const Toggletip = React.memo(
+  ({
+    children,
+    theme = 'info',
+    placement = 'auto',
+    content,
+    title,
+    closeButton = true,
+    onClose,
+    footer,
+  }: ToggletipProps) => {
+    const styles = useStyles2(getStyles);
+    const containerStyle = styles[theme];
+    const contentRef = useRef(null);
+    const [controlledVisible, setControlledVisible] = React.useState(false);
+
+    useEffect(() => {
+      if (controlledVisible) {
+        const handleKeyDown = (enterKey: KeyboardEvent) => {
+          if (enterKey.key === 'Escape') {
+            setControlledVisible(false);
+            if (!!onClose) {
+              onClose();
+            }
+          }
+        };
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+          document.removeEventListener('keydown', handleKeyDown);
+        };
+      } else if (controlledVisible === false && !!onClose) {
+        onClose();
+      }
+      return;
+    }, [controlledVisible, onClose]);
+
+    const { getArrowProps, getTooltipProps, setTooltipRef, setTriggerRef, visible, update } = usePopperTooltip({
+      visible: controlledVisible,
+      placement: placement,
+      interactive: true,
+      offset: [0, 8],
+      trigger: 'click',
+      onVisibleChange: setControlledVisible,
+    });
+
+    return (
+      <>
+        {React.cloneElement(children, {
+          ref: setTriggerRef,
+          tabIndex: 0,
+        })}
+        {visible && (
+          <Portal>
+            <div
+              data-testid="toggletip-content"
+              ref={setTooltipRef}
+              {...getTooltipProps({ className: containerStyle })}
+            >
+              {!!title && (
+                <div className="toggletip-header">
+                  {typeof title === 'string' && title}
+                  {title && React.isValidElement(title) && React.cloneElement(title)}
+                </div>
+              )}
+              {closeButton && (
+                <div data-testid="toggletip-header-close" className="toggletip-header-close">
+                  <IconButton
+                    aria-label="Close Toggletip"
+                    name="times"
+                    size="md"
+                    onClick={() => setControlledVisible(false)}
+                  />
+                </div>
+              )}
+              <div ref={contentRef} {...getArrowProps({ className: 'toggletip-arrow' })} />
+              <div className="toggletip-body">
+                {typeof content === 'string' && content}
+                {React.isValidElement(content) && React.cloneElement(content)}
+                {typeof content === 'function' && update && content({ update })}
+              </div>
+              {!!footer && (
+                <div className="toggletip-footer">
+                  {typeof footer === 'string' && footer}
+                  {footer && React.isValidElement(footer) && React.cloneElement(footer)}
```

Comment:
Question: is this necessary? Would be simply just be able to render the `{footer}` instead of checking the different types? Or is this added as a runtime check if the consumer isn't using typescrip?

Reply:
To be honest, I think the type check is not necessary. As I don't need to extend the properties of the received component, the cloneElement is not necessary. So, I have removed the check on header and footer and just rendered the component or string as It is.